tests/foreman/virtwho/utils.py

```diff
@@ -131,6 +133,8 @@ def virtwho_cleanup():
     runcmd("rm -f /var/run/virt-who.pid")
     runcmd("rm -f /var/log/rhsm/rhsm.log")
     runcmd("rm -rf /etc/virt-who.d/*")
+    for host in Host.list():
+        Host.delete({'id': host['id']})
```

Comment:
Just to confirm - these tests are running on separate Satellite, right? There is no risk of this part interfering with our (Sat QE) automation, correct?

Overall, removing all registered hosts doesn't sound like a good idea for me, but if you have full control over that environment and believe this is the best approach, I'm OK with that.

Reply:
@mirzal Yes, there is no influence for the SatQE automation, we deploy the satellite by virtwho jenkins and only used to test virtwho-configure cases.

Reply:
Although our satellite is a separated server, I still agree with you it's not a good idea to remove all the hosts, imaging that someone wants to use his satellite server to debug these cases, his original hosts will be deleted, if that, it's too bad.

Why I need to delete the host which sent by virt-who:
When I use hammer to list a specific subscription, it will return two or more items, the redundant
items is caused by virt-who have consumed the physical subscription and generate the virtual bonus subscription for guests, such as:
```
# hammer -v -u admin -p admin  --output=csv subscription list --organization="Default Organization" --search="product_id=RH00001"
ID,UUID,Name,Type,Contract,Account,Support,End Date,Quantity,Consumed
33,8ac213ac6cdbced2016cfac546da0d45,"Red Hat Enterprise Linux for Virtual Datacenters, Premium",Guests of virt-who-bootp-73-131-242.rhts.eng.pek2.redhat.com-1,...
32,8ac213ac6cdbced2016cfab1031c0d39,"Red Hat Enterprise Linux for Virtual Datacenters, Premium",Guests of virt-who-win-virt-who-ci-1,...
5,8ac213ac6cdbced2016cdbeb606d0814,"Red Hat Enterprise Linux for Virtual Datacenters, Premium",Physical,...
```
So I found another solution to fix this issue as below:
```
            host = Host.list({'search': hostname})[0]
            items = Subscription.list({
                'organization': DEFAULT_ORG,
                'search': sku,
            })
            vdc_id = items[0]['id']
            if sku == 'type=STACK_DERIVED':
                for item in items:
                    if hypervisor_name in item['type']:
                        vdc_id = item['id']
                        break
            result = Host.subscription_attach({
                'host-id': host['id'],
                'subscription-id': vdc_id
            })
```
When `type=STACK_DERIVED`, it will return two or more subscriptions, I need the item which including `hypervisor_name`.